### PR TITLE
New OS docs

### PR DESCRIPTION
**Pull request by Pezi777**

### CODE REVIEW

File: _includes/c11-cpp11-and-beyond-and-toolchains.md [hunk 1]
```diff
@@ -204,12 +221,30 @@ matrix:
       
 You can find the `clang` version shipped by Xcode [here](https://trac.macports.org/wiki/XcodeVersionInfo).
 
+### Clang on FreeBSD
+
+FreeBSD ships with Clang 8.0.1
```
Comment (ararslan) [inline _includes/c11-cpp11-and-beyond-and-toolchains.md]: Would this be an appropriate place to mention that the default compiler on FreeBSD is Clang? People not familiar with FreeBSD might assume it's GCC like on Linux.

File: user/reference/freebsd.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+## Go support
+
+* Pre-installed Go: 1.11
+* Other Go versions can be installed during build time by specifying the language versions with the go:-key.
+
+
+## Julia support
+
+* Supported version TBD
```
Comment (ararslan) [inline user/reference/freebsd.md]: Julia supports FreeBSD as of Julia version 0.7, so any version >= 0.7 should work in the same manner as the other platforms.

File: _includes/c11-cpp11-and-beyond-and-toolchains.md (hunk 1 above)
Comment (Pezi777) [reply _includes/c11-cpp11-and-beyond-and-toolchains.md]: sure, will add this here :-)

**Review by ararslan — DISMISSED** [review]

File: _includes/c11-cpp11-and-beyond-and-toolchains.md [hunk 3]
```diff
@@ -204,12 +221,34 @@ matrix:
...
+
+To upgrade Clang to a more recent version, install the appropriate version from packages; see below for examples:
+
+```yaml
+os: freebsd
+addons:
+    pkg:
+      - llvm90
```
Comment (lwhsu) [inline _includes/c11-cpp11-and-beyond-and-toolchains.md:236]: maybe we can use `llvm10` as example now, btw, `llvm11` is also available.

File: _includes/c11-cpp11-and-beyond-and-toolchains.md [hunk 4]
```diff
@@ -204,12 +221,34 @@ matrix:
...
+    env:
+      - CC=clang
+      - CXX=clang++
```
Comment (lwhsu) [inline _includes/c11-cpp11-and-beyond-and-toolchains.md:239]: `llvm90` installs these tools as `/usr/local/bin/clang90` and `/usr/local/bin/clang++90`.

File: user/languages/java.md [hunk 5]
```diff
@@ -232,6 +232,8 @@ addons:
 ```
 {: data-file=".travis.yml"}
 
+> FreeBSD does not have pre-installed Oracle JDK 8. Due to the current licensing policy, Oracle JDK on FreeBSD binaries can not be distributed and you are only permitted to use them personally. For the same reason, the sources must be fetched manually. For more FreeBSD Oracle Java support information and available FreeBSD ports, see [here](https://www.freebsd.org/java/).
+
```
Comment (lwhsu) [inline user/languages/java.md:236]: Should we mention that FreeBSD has `openjdk*` pacakges?

File: _includes/c11-cpp11-and-beyond-and-toolchains.md [hunk 6]
```diff
@@ -98,6 +98,23 @@ before_install:
 ```
 {: data-file=".travis.yml"}
 
+### GCC on FreeBSD
+
+FreeBSD ships with GCC 10.0.0
```
Comment (lwhsu) [inline _includes/c11-cpp11-and-beyond-and-toolchains.md:103]: Should we note that it is Travis-CI's FreeBSD image has GCC 10 pre-installed. It's not in the base system by default.

File: user/languages/python.md [hunk 7]
```diff
@@ -170,6 +170,9 @@ jobs:
   include:
     - name: "Python 3.8.0 on Xenial Linux"
       python: 3.8           # this works for Linux but is ignored on macOS or Windows
+    - name: "Python 3.6.10 on FreeBSD"
```
Comment (lwhsu) [inline user/languages/python.md:173]: The current default python version on FreeBSD is 3.7, maybe we want to rebuild the image and update this in the future?

File: user/reference/freebsd.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+* Default version: 8
+* Pre-installed OpenJDK version 8, 11, 12, 13 (OpenJDK10 not supported in FreeBSD)
+* Pre-installed Apache Ant(TM) version 1.10.6
+* Pre-installed Apache Maven version 3.6.3
+* Pre-installed Gradle version 6.0.1
+
+## Docker
+
+Currently unsupported. See [FreeBSD wiki](https://wiki.freebsd.org/Docker) for more details.
```
Comment (lwhsu) [inline user/reference/freebsd.md:121]: And maybe refer to https://wiki.freebsd.org/ContainerOrchestration for more native solutions.

File: user/installing-dependencies.md [hunk 9]
```diff
@@ -303,6 +303,26 @@ rvm use $TRAVIS_RUBY_VERSION # optionally, switch back to the Ruby version you n
 
 > You can also have a look at the [Homebrew](https://config.travis-ci.com/ref/job/addons/homebrew) section in our [Travis CI Build Config Reference](https://config.travis-ci.com/).
 
+## Installing Packages on FreeBSD
+
+To install packages that are not included in the default FreeBSD environment use `pkg` in the `before_install` step of your `.travis.yml`:
+
+```yaml
+before_install:
+  - sudo pkg install -y curl
```
Comment (tanzislam) [inline user/installing-dependencies.md:312]:
```suggestion
  - su -m root -c 'pkg install -y curl'
```
FreeBSD [doesn't ship](https://docs.freebsd.org/en/books/handbook/security/#security-sudo) with the `sudo` utility. It wasn't present in the Travis image when I checked back in January 2021; not sure if Travis includes it now.

**Review by michal-at-travisci — APPROVED** [review]

**Pull request merged**